framework/tlvf/src/src/CmduMessageTx.cpp

```diff
@@ -77,42 +56,5 @@ std::shared_ptr<tlvVendorSpecific> CmduMessageTx::add_vs_tlv(tlvVendorSpecific::
 
 bool CmduMessageTx::finalize(bool swap_needed)
 {
-    if (m_finalized)
-        return true;
-
-    if (!m_cmdu_header)
-        return false; 
-
-    // Update vendor specific length which is seperated to 3 classes
```

Comment:
As I mentioned in the daily: if you don't find a better solution, you can just keep this hack for the time being. Allocate the vendorspecific TLV with a minimum size (which is how it was done before) and overwrite the length field like below.

Reply:
Did it a bit differently, but it works.